components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/FederatedAuthenticatorConfig.java

```diff
@@ -230,4 +246,24 @@ public void setTags(String[] tagList) {
 
         tags = tagList;
     }
+
+    /**
+     * Get the tag list of the Local authenticator.
+     *
+     * @return String[]
+     */
+    public IdentityConstants.DefinedByType getDefinedByType() {
```

Comment:
Can we have this DefinedByType enum defined relative to application authenticator than using IdentityConstants

Reply:
We cannot do this, as it cause circular dependency.

Reply:
Then still make sure this is related to authenticator configs. You can have a different class to denote constants related to authenticators